pkg/ebpf/perf.go

```diff
@@ -44,30 +47,33 @@ var recordPool = sync.Pool{
 }
 
 // NewPerfHandler creates a PerfHandler
-func NewPerfHandler(dataChannelSize int) *PerfHandler {
+func NewPerfHandler(dataChannelSize int, mapName string) *PerfHandler {
```

Comment:
Doesn't seem intuitive to pass in the map name here. Did you consider moving collecting the metrics to ebpf-manager?

Reply:
Updated to a manager-based method, please re-review.